include/boost/geometry/algorithms/detail/overlay/intersection_insert.hpp

```diff
@@ -380,12 +379,10 @@ struct intersection_of_linestring_with_areal
         }
         
 #if defined(BOOST_GEOMETRY_DEBUG_FOLLOW)
-        int index = 0;
-        for(typename std::deque<turn_info>::const_iterator
-            it = turns.begin(); it != turns.end(); ++it)
-        {
-            debug_follow(*it, it->operations[0], index++);
-        }
+        for_each_with_index(turns, [](auto index, auto const& turn)
+            {
+                debug_follow(turn, turn.operations[0], index);
+            });
```

Comment:
The indentation is inconsistent with other calls. Besides we should probably mimic loops as it is correctly done in other places.

Reply:
:heavy_check_mark: